projects/graph/src/graph.py

```diff
@@ -2,8 +2,32 @@
 Simple graph implementation compatible with BokehGraph class.
 """
 
-
+import random
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices={}
+    def add_vertex(self,vertex_id):
+        self.vertices[vertex_id]=Vertex(vertex_id)
+    def add_edge(self,v1,v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].edges.add(v2)
+            self.vertices[v2].edges.add(v1)
+        else:
+            raise IndexError('That vertex does not exist.')
+        
+
+class Vertex:
+    def __init__(self,vertex_id,x=None,y=None):
```

Comment:
You could add self.color to vertex and randomly generate a color for each.  Then override that color when you get to part 5 of this repo.